typescript/web/src/components/import-button/import-images-modal/modal-dropzone/import-dropped-files/import-images.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable no-underscore-dangle */
+import { ApolloClient, gql } from "@apollo/client";
+import { v4 as uuidv4 } from "uuid";
+import Bluebird from "bluebird";
+import mime from "mime-types";
+import chunk from "lodash/fp/chunk";
+
+import { uploadFile } from "../../../../../utils/upload-file";
+import { DroppedFile, SetUploadStatuses } from "../../types";
+
+import { BATCH_SIZE, CONCURRENCY } from "../../constants";
+
+const createManyImagesMutation = gql`
+  mutation ($images: [ImageCreateManySingleImage!]!, $datasetId: ID!) {
+    createManyImages(data: { images: $images, datasetId: $datasetId }) {
+      id
+    }
+  }
+`;
+
+const uploadBatchOfImages = async ({
+  batch,
+  firstUploadDate,
+  batchIndex,
+  datasetId,
+  workspaceId,
+  apolloClient,
+}: {
+  batch: DroppedFile[];
+  firstUploadDate: Date;
+  batchIndex: number;
+  datasetId: string;
+  workspaceId: string;
+  apolloClient: ApolloClient<any>;
+}) => {
+  return await Promise.all(
+    batch.map(async ({ file }, fileIndex) => {
+      const createdAtDate = new Date();
+      createdAtDate.setTime(
+        firstUploadDate.getTime() + batchIndex * BATCH_SIZE + fileIndex
+      );
+      const createdAt = createdAtDate.toISOString();
```

Comment:
Why to we need this?

Reply:
The image order is based on their created at. 
To keep the order given during the upload, we artificially add 1ms between 2 images. (That how it was done so far, it's not a behaviour I added in this PR)